certified-connectors/DocuSignDemo/script.csx

```diff
@@ -1614,6 +1624,97 @@ public class Script : ScriptBase
       response.Content = new StringContent(body.ToString(), Encoding.UTF8, "application/json");
     }
 
+    if ("GetRecipientFields".Equals(this.Context.OperationId, StringComparison.OrdinalIgnoreCase))
+    {
+      var body = ParseContentAsJObject(await response.Content.ReadAsStringAsync().ConfigureAwait(false), false);
+      List<JToken> recipientList = new List<JToken>();
+      recipientList.Add(body["signers"]);
+      recipientList.Add(body["agents"]);
+      recipientList.Add(body["editors"]);
+      recipientList.Add(body["carbonCopies"]);
+      recipientList.Add(body["certifiedDeliveries"]);
+      recipientList.Add(body["intermediaries"]);
+      recipientList.Add(body["inPersonSigners"]);
+      recipientList.Add(body["seals"]);
+      recipientList.Add(body["witnesses"]);
+      recipientList.Add(body["notaries"]);
+
+      var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+
+      var newSigner = new JObject();
+      newSigner["signer"] = null;
+      var newBody = new JObject();
+      var recipientEmailId = query.Get("recipientEmail");
+      var phoneNumber = query.Get("areaCode") + " " + query.Get("phoneNumber");
+      var signerPhoneNumber = "";
+      char[] charsToTrimPhoneNumber = { '*', ' ', '\'', '/', '-', '(', ')', '+'};
+
+      if (recipientEmailId != null)
+      {
+        for(var i = 0; i < recipientList.Count; i++)
+        {
+          foreach(var signer in recipientList[i])
+          {
+            if (recipientEmailId.ToString().Equals(signer["email"].ToString()))
+            {
+              newSigner["signer"] = signer;
```

Comment:
Earlier you were missing a typecast, so it was not working.
`matchingSigner = signer as JObject;`
Naming to matchingSigner seems more appropriate than newSigner

Reply:
Updated the code with the suggestion.